### PR TITLE
Patch csv-generator to workaround an OLM bug

### PR DESCRIPTION
**Pull request by tiraboschi**
**Release note**:
```release-note
patch csv-generator to workaround an OLM bug
```

### CODE REVIEW

File: tools/csv-merger/csv-merger.go [hunk 1]
```diff
@@ -333,6 +333,13 @@ func main() {
 
 			strategySpec := csvStruct.Spec.InstallStrategy.StrategySpec
 
+			// temporary workarounf for RHBZ#....
```
Comment (zcahana) [inline tools/csv-merger/csv-merger.go:336]: workarounf --> workaround

Comment (zcahana) [inline tools/csv-merger/csv-merger.go:336]: Is there actually a bugzilla on this? If so, it's worthwhile to link it so a future maintainer will have the context for why it was added (and when can be removed).

File: tools/csv-merger/csv-merger.go [hunk 2]
```diff
@@ -333,6 +333,13 @@ func main() {
...
+			// a custom .spec.template.annotations["description"] it's causing a failure on OLM
+			// TODO: remove this once fixed on OLM side
+			for _, deployment := range strategySpec.DeploymentSpecs {
+				delete(deployment.Spec.Template.Annotations, "description")
```
Comment (zcahana) [inline tools/csv-merger/csv-merger.go:340]: This handles a very specific annotation ("description").
Should we instead mimic the [OLM logic](https://github.com/operator-framework/operator-lifecycle-manager/blob/4b66803055a8ab611447c33ed86e755ad39cb313/pkg/controller/install/deployment.go#L263-L269) and just mirror the annotations from the CSV?

File: tools/csv-merger/csv-merger.go (hunk 1 above)
Comment (tiraboschi) [reply tools/csv-merger/csv-merger.go:336]: done

File: tools/csv-merger/csv-merger.go (hunk 2 above)
Comment (tiraboschi) [reply tools/csv-merger/csv-merger.go:340]: we have more than one deployment in the CSV,
replicating the same long description more than once it's just going to make it unnecessarily long without any real value

Comment (zcahana) [reply tools/csv-merger/csv-merger.go:340]: True, there's no value in replicating the same description over and over.

**Pull request merged**